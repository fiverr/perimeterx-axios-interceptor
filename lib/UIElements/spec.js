const {
    CHALLENGE_BOX_CLASSNAME,
    MODAL_CLASSNAME,
    QUICKFIX_CLASSNAME,
    SUBTITLE_CLASSNAME,
    TITLE_CLASSNAME
} = require('./styles');
const { removeCaptcha, createModal } = require('.');

describe('lib/UIElements', () => {
    afterEach(() => {
        removeCaptcha();
    });
    describe('removeCaptcha', () => {
        it('should find and remove any existing PerimeterX challenges on the page', () => {
            const pxCaptcha = document.createElement('div');
            pxCaptcha.id = 'px-captcha';
            document.body.appendChild(pxCaptcha);

            const modal = document.createElement('dialog');
            modal.className = MODAL_CLASSNAME;
            document.body.appendChild(modal);

            removeCaptcha();
            expect(
                document.body.querySelector('#px-captcha')
            ).toBe(null);
            expect(
                document.body.querySelector(`.${MODAL_CLASSNAME}`)
            ).toBe(null);
        });
    });
    describe('createModal', () => {
        it('should build the modal UI, PerimeterX and default content', () => {
            expect(createModal()).toMatchSnapshot();
        });
        it('should include the challenge box', () => {
            const modal = createModal();
            const challenge = modal.querySelector(`.${CHALLENGE_BOX_CLASSNAME}`);
            expect(challenge.id).toBe('px-captcha');

        });
        it('should accept custom className', () => {
            expect(createModal().className).toBe(MODAL_CLASSNAME);
            expect(createModal({ className: 'customised' }).className).toBe('customised');
        });
        it('should use title override', () => {
            const selector = `p.${TITLE_CLASSNAME}`;
            expect(createModal().querySelector(selector)).not.toBe(null);
            expect(createModal({ title: null }).querySelector(selector)).toBe(null);
            expect(createModal({ title: 'Something Unexpected' }).querySelector(selector).textContent).toBe('Something Unexpected');
        });
        it('should use subtitle override', () => {
            const selector = `p.${SUBTITLE_CLASSNAME}`;
            expect(createModal().querySelector(selector)).not.toBe(null);
            expect(createModal({ subtitle: null }).querySelector(selector)).toBe(null);
            expect(createModal({ subtitle: 'Something Unexpected' }).querySelector(selector).textContent).toBe('Something Unexpected');
        });
        it('should use contactSupport override', () => {
            const selector = 'p[class=""]';
            expect(createModal().querySelector(selector)).not.toBe(null);
            expect(createModal({ contactSupport: null }).querySelector(selector)).toBe(null);
            expect(createModal({ contactSupport: 'Something Unexpected' }).querySelector(selector).textContent).toBe('Something Unexpected');
        });
        it('should use quickfixes override', () => {
            const selector = `p[class="${QUICKFIX_CLASSNAME}"]`;
            expect(createModal().querySelector(selector)).not.toBe(null);
            expect(createModal({ quickfixes: null }).querySelector(selector)).toBe(null);

            const list = createModal({ quickfixes: ['a. 1', 'b. 2'] }).querySelectorAll(selector);
            expect(list.length).toBe(2);
            expect([].map.call(list, ({textContent}) => textContent)).toEqual(['a. 1', 'b. 2']);
        });
    });
});
