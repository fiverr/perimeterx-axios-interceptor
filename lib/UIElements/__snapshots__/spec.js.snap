// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/UIElements createModal should build the modal UI, PerimeterX and default content 1`] = `
<dialog
  class="perimeterx-async-challenge"
  open=""
>
  <div>
    <p
      class="title"
    >
      One Small Step
    </p>
    <p
      class="subtitle"
    >
      Please check the box below to continue your normal visit
    </p>
    <div
      class="challenge-box"
      id="px-captcha"
    />
    <p
      class="quickfix"
    >
      Please exclude this website from ad blocking or ad filtering software.
    </p>
    <p
      class="quickfix"
    >
      Make sure you don't have any browser extensions tampering with request headers or user agent string.
    </p>
    <p
      class="quickfix"
    >
      Make sure JavaScript is enabled in your browser.
    </p>
    <p>
      If you're still having trouble accessing the site, please contact customer support.
    </p>
    <style>
      
.perimeterx-async-challenge {
    z-index: 9999;
    position: fixed;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    border: 0;
    margin: 0;
    padding: 0;
    background: rgba(0, 0, 0, .3);
    color: black;
}
.perimeterx-async-challenge &gt; div {
    margin: 20vh 20vw 0;
    background: white;
    box-shadow: 0 0 2em rgba(0, 0, 0, .4);
    padding: 1em 1.5em;
}
.perimeterx-async-challenge p,
.perimeterx-async-challenge .challenge-box {
    margin: 0 0 .5em;
}
.perimeterx-async-challenge .title {
    font-size: 2em;
    font-weight: bold;
}
.perimeterx-async-challenge .subtitle {
    font-size: 1.4em;
}
.perimeterx-async-challenge .quickfix {
    font-size: .8em;
    margin: 0;
}
.perimeterx-async-challenge .quickfix:before {
    content: "â€¢";
    margin: 0 .5em
}
@media screen and (max-width:1040px) {
    .perimeterx-async-challenge &gt; div {
        margin: 10vh 10vw 0;
    }
}
@media screen and (max-width:800px) {
    .perimeterx-async-challenge &gt; div {
        margin: 5vw 5vw 0;
    }
}

    </style>
  </div>
</dialog>
`;
